jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -931,7 +940,15 @@ private static void checkPermsNPE(Permission... perms) {
 	checkPermsNPE(perms);
 	ProtectionDomain domain = getCallerPD(1);
 	ProtectionDomain[] pdArray = (domain == null) ? null : new ProtectionDomain[] { domain };
-	return doPrivileged(action, new AccessControlContext(context, pdArray, getNewAuthorizedState(context, domain)), perms);
+	AccessControlContext computedContext = getContextHelper(true);
```

Comment:
getContextHelper() should only be called if necessary. There is a path where it's not needed. Same in the other place.

Reply:
Changed the logic so `getContextHelper` is only called if the passed context is null and we don't try to set the context's domain combiner anymore.

The HttpNegotiateServer test still passes with the change.

Reply:
`getCallerPD(1)` is not needed w/ usage of `getContextHelper()` which contains the caller PD.

Reply:
Changed it so that the `null == context` path just calls `getContextHelper(true)` and the other path uses the ProtectionDomain from `getCallerPD`.